src/platformdirs/__init__.py

```diff
@@ -19,18 +19,18 @@
     from pathlib import Path
     from typing import Literal
 
+if sys.platform == "win32":
+    from platformdirs.windows import Windows as _Result
+elif sys.platform == "darwin":
+    from platformdirs.macos import MacOS as _Result
+else:
+    from platformdirs.unix import Unix as _Result
 
-def _set_platform_dir_class() -> type[PlatformDirsABC]:
-    if sys.platform == "win32":
-        from platformdirs.windows import Windows as Result  # noqa: PLC0415
-    elif sys.platform == "darwin":
-        from platformdirs.macos import MacOS as Result  # noqa: PLC0415
-    else:
-        from platformdirs.unix import Unix as Result  # noqa: PLC0415
 
+def _set_platform_dir_class() -> type[PlatformDirsABC]:
```

Comment:
Note that if https://github.com/python/mypy/issues/10962 is fixed, this should become
```suggestion
def _set_platform_dir_class() -> type[_Result | Android]:
```

But that'd require duplicating the `from platformdirs.android import Android` import in the `TYPE_CHECKING` block to keep it a lazy import. And it brings no benefit at the current time.